docs/api.md

```diff
@@ -118,8 +120,93 @@ type AddonsSpec struct {
+	MaxPodGracePeriod *int32 `json:"maxPodGracePeriod,omitempty"`
+
+	// Maximum time CA waits for node to be provisioned
+	// +kubebuilder:validation:Pattern=^([0-9]+(\.[0-9]+)?(ns|us|µs|ms|s|m|h))+$
+	MaxNodeProvisionTime string `json:"maxNodeProvisionTime,omitempty"`
+
+	// To allow users to schedule "best-effort" pods, which shouldn't trigger
+	// Cluster Autoscaler actions, but only run when there are spare resources available,
+	// More info: https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#how-does-cluster-autoscaler-work-with-pod-priority-and-preemption
+	PodPriorityThreshold *int32 `json:"podPriorityThreshold,omitempty"`
+
+	// BalanceSimilarNodeGroups enables/disables the
+	// `--balance-similar-node-groups` cluster-autocaler feature.
+	// This feature will automatically identify node groups with
+	// the same instance type and the same set of labels and try
+	// to keep the respective sizes of those node groups balanced.
+	BalanceSimilarNodeGroups *bool `json:"balanceSimilarNodeGroups,omitempty"`
+
+	// Enables/Disables `--ignore-daemonsets-utilization` CA feature flag. Should CA ignore DaemonSet pods when calculating resource utilization for scaling down. false by default
+	IgnoreDaemonsetsUtilization *bool `json:"ignoreDaemonsetsUtilization,omitempty"`
+
+	// Enables/Disables `--skip-nodes-with-local-storage` CA feature flag. If true cluster autoscaler will never delete nodes with pods with local storage, e.g. EmptyDir or HostPath. true by default at autoscaler
+	SkipNodesWithLocalStorage *bool `json:"skipNodesWithLocalStorage,omitempty"`
+}
+
+type ResourceLimits struct {
+	// Maximum number of nodes in all node groups.
+	// Cluster autoscaler will not grow the cluster beyond this number.
+	// +kubebuilder:validation:Minimum=0
+	MaxNodesTotal *int32 `json:"maxNodesTotal,omitempty"`
+
+	// Minimum and maximum number of cores in cluster, in the format <min>:<max>.
+	// Cluster autoscaler will not scale the cluster beyond these numbers.
+	Cores *ResourceRange `json:"cores,omitempty"`
+
+	// Minimum and maximum number of gigabytes of memory in cluster, in the format <min>:<max>.
+	// Cluster autoscaler will not scale the cluster beyond these numbers.
+	Memory *ResourceRange `json:"memory,omitempty"`
+
+	// Minimum and maximum number of different GPUs in cluster, in the format <gpu_type>:<min>:<max>.
+	// Cluster autoscaler will not scale the cluster beyond these numbers. Can be passed multiple times.
+	GPUS []GPULimit `json:"gpus,omitempty"`
+}
+
+type GPULimit struct {
+	// +kubebuilder:validation:MinLength=1
+	Type string `json:"type"`
+
+	// +kubebuilder:validation:Minimum=0
+	Min int32 `json:"min"`
+	// +kubebuilder:validation:Minimum=1
+	Max int32 `json:"max"`
+}
+
+type ResourceRange struct {
+	// +kubebuilder:validation:Minimum=0
+	Min int32 `json:"min"`
+	Max int32 `json:"max"`
+}
+
+type ScaleDownConfig struct {
+	// Should CA scale down the cluster
+	Enabled bool `json:"enabled"`
```

Comment:
The scale down config itself is an optional object. If an omitted scale down config is interpreted as "scale down disabled" (I'm assuming — this needs to be specified), what is the purpose of specifying the field but declaring it enabled=false vs. omitting the field? Would it be possible to remove this field entirely?